Ast/src/Parser.cpp

```diff
@@ -2615,6 +2639,72 @@ AstExpr* Parser::parseString()
         return reportExprError(location, {}, "String literal contains malformed escape sequence");
 }
 
+AstExpr* Parser::parseInterpString()
+{
+    std::vector<AstArray<char>> strings;
+    std::vector<AstExpr*> expressions;
+
+    Location startLocation = lexer.current().location;
+
+    do {
+        auto currentLexeme = lexer.current();
```

Comment:
```suggestion
        Lexeme currentLexeme = lexer.current();
```

Reply:
Resolving this. What are the rules of auto in the codebase?

Reply:
Basically we almost never use `auto` unless it's in context of a ranged-for, destructuring (where auto is required by C++), or if the type is explicitly spelled out on the right such as static_cast<> or as<> (although even then we try to avoid it unless it gets too repetitive)